agent/src/flow_generator/protocol_logs/http.rs

```diff
@@ -1649,8 +1650,14 @@ impl<'a> Iterator for V1HeaderIterator<'a> {
             None
         } else {
             let result = unsafe {
-                // this is safe because all bytes are checked to be ascii
-                str::from_utf8_unchecked(&self.0[..end])
+                if is_ascii {
+                    str::from_utf8_unchecked(&self.0[..end])
+                } else {
+                    match str::from_utf8(&self.0[..end]) {
+                        Err(_) => return None,
+                        Ok(s) => s,
+                    }
+                }
```

Comment:
```suggestion
                if is_ascii {
                    // this is safe because all bytes are checked to be ascii
                    str::from_utf8_unchecked(&self.0[..end])
                } else {
                    str::from_utf8(&self.0[..end]).ok()
                }
```

Reply:
再调整一下 unsafe 的位置